akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/voice/SetUserListenOnlyInputCmdMsgHdlr.scala

```diff
@@ -0,0 +1,54 @@
+package org.bigbluebutton.core.apps.voice
+
+import org.bigbluebutton.common2.msgs.SetUserListenOnlyInputCmdMsg
+import org.bigbluebutton.core.apps.{ PermissionCheck, RightsManagementTrait }
+import org.bigbluebutton.core.apps.voice.VoiceApp
+import org.bigbluebutton.core.models.{ Roles, Users2x, VoiceUsers }
+import org.bigbluebutton.core.running.{ BaseMeetingActor, LiveMeeting, OutMsgRouter }
+import org.bigbluebutton.core2.MeetingStatus2x
+
+trait SetUserListenOnlyInputCmdMsgHdlr extends RightsManagementTrait {
+  this: BaseMeetingActor =>
+
+  val liveMeeting: LiveMeeting
+  val outGW: OutMsgRouter
+
+  def handleSetUserListenOnlyInputCmdMsg(msg: SetUserListenOnlyInputCmdMsg) {
+    if (msg.body.userId != msg.header.userId && (permissionFailed(
+      PermissionCheck.MOD_LEVEL,
+      PermissionCheck.VIEWER_LEVEL, liveMeeting.users2x, msg.header.userId
+    ))) {

```

Comment:
I'm not sure this validation is appropriate.
As I see it, only the user themselves should be allowed to set this flag, so checking the Role doesn't seem relevant in this case.